repository/gridftp/globus5/authz/tests/unit/test_AuthzWorker.cpp

```diff
@@ -0,0 +1,409 @@
+#define BOOST_TEST_MAIN
+
+#define BOOST_TEST_MODULE AuthzWorker
+#include <boost/filesystem.hpp>
+#include <boost/test/unit_test.hpp>
+
+// Private includes
+#include "AuthzWorker.hpp"
+
+// Public includes
+#include "common/DynaLog.hpp"
+#include "common/ICommunicator.hpp"
+#include "common/IMessage.hpp"
+#include "common/MessageFactory.hpp"
+#include "common/SDMS_Anon.pb.h"
+#include "common/TraceException.hpp"
+
+extern "C" {
+#include "Config.h"
+}
+
+class ConfigFixture {
+public:
+  // Setup: create the object before each test case
+  // The below settings are simply for testing and are not 
+  // used in anyway in a production environment.
+  ConfigFixture() {
+    std::string repo_id = "datafed-one-repo-to-rule-them-all";
+    std::string server_addr = "tcp://ruler:7513";
+    std::string pub_key = "=+r^&Qc&}f<Pho7ViJwVM5(Ze^o%UnnyA:XjF}7Q";
+    std::string priv_key = "X4y+AFXCmBoyKrc0J0V5}Z2HpCYRf3(wvhe**z))";
+    std::string server_key = "a.r)OeoL=?/rHaK1-ow<xOd7YxM/6)A&Kdu]*<)3";
+    std::string user = "vader";
+    std::string test_path = "";
+    std::string log_path = "./";
+    std::string globus_collection_path = "/globus/root";
+    //
+    std::strcpy(config.repo_id, repo_id.c_str());
+    std::strcpy(config.server_addr, server_addr.c_str());
+    std::strcpy(config.pub_key, pub_key.c_str());
+    std::strcpy(config.priv_key, priv_key.c_str());
+    std::strcpy(config.server_key, server_key.c_str());
+    std::strcpy(config.user, user.c_str());
+    std::strcpy(config.test_path, test_path.c_str());
+    std::strcpy(config.log_path, log_path.c_str());
+    std::strcpy(config.globus_collection_path, globus_collection_path.c_str());
+  }
+
+  // Teardown: clean up after each test case
+  ~ConfigFixture() {}
+
+  // Accessor for the object
+
+  Config config;
+
+private:
+};
+
+BOOST_FIXTURE_TEST_SUITE(AuthzTest, ConfigFixture)
+
+BOOST_AUTO_TEST_CASE(test_authz) {
+  SDMS::LogContext log_context;
+  SDMS::AuthzWorker worker(&config, log_context);
+  std::cout << std::string(config.repo_id) << std::endl;
```

Comment:
Is the point of this test to validate an error does not occur during worker construction?

Reply:
Yes, the test name could be improved.